conwayste/src/error.rs

```diff
@@ -0,0 +1,45 @@
+/*  Copyright 2019 the Conwayste Developers.
+ *
+ *  This file is part of Conwayste.
+ *
+ *  Conwayste is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Conwayste is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with libconway.  If not, see <http://www.gnu.org/licenses/>. */
+
+custom_error! {pub ConwaysteError
+    InvalidUIAction {reason: String} = "ConwaysteError->InvalidUIAction->{reason}",
+    NoAssociatedUIAction{reason: String} = "ConwaysteError->NoAssociatedUIAction->{reason}"
+}
+
+pub type ConwaysteResult<T> = ::std::result::Result<T, ConwaysteError>;
+
+impl PartialEq for ConwaysteError {
```

Comment:
Is this needed?

Reply:
Are you referring to the `PartialEq` implementation or the general need for a `ConwaysteError`? For the former I followed the `ConwayError` use case so that the same level of support is met.

Reply:
This compiles without `PartialEq`. I don't see a good reason to keep it.

Reply:
Indeed it does. We should also remove it for `ConwayError` too then. Removed in one of the newer commits.